packages/nuxt/src/pages/plugins/page-meta.ts

```diff
@@ -122,32 +122,33 @@ export const PageMetaPlugin = createUnplugin((options: PageMetaPluginOptions) =>
           const name = 'name' in node.callee && node.callee.name
           if (name !== 'definePageMeta') { return }
 
-          const meta = node.arguments[0] as Expression & { start: number, end: number }
-
-          let contents = `const __nuxt_page_meta = ${code!.slice(meta.start, meta.end) || 'null'}\nexport default __nuxt_page_meta` + (options.dev ? CODE_HMR : '')
-
-          function addImport (name: string | false) {
-            if (name && importMap.has(name)) {
-              const importValue = importMap.get(name)!.code
-              if (!addedImports.has(importValue)) {
-                contents = importMap.get(name)!.code + '\n' + contents
-                addedImports.add(importValue)
+          const meta = node.arguments[0] as (Expression & { start: number, end: number }) | undefined
```

Comment:
This plugin should already be doing it:

https://github.com/nuxt/nuxt/blob/ff1bb56e3f90472ff7aae69b214e21197c78400b/packages/nuxt/src/core/plugins/tree-shake.ts#L29-L46

It would be worth investigating why it is not in this case.